app/assets/javascripts/lib/util/empty_chart_message.js

```diff
@@ -0,0 +1,39 @@
+var EmptyChartMessage = (function () {
+    'use strict';
+
+    var template = function (container) {
+        var wrapper,
+            existingWrapper = container.find(".empty-wrapper")
+
+        if (existingWrapper.length > 0) {
+            wrapper = existingWrapper;
+        } else {
+            wrapper = $("<div>").addClass("empty-wrapper");
+            wrapper.append(
+                $("<p>").text(I18n.t("output_elements.common.empty")),
+                $("<div>").text(I18n.t("output_elements.empty." + this.key)).addClass("anotation")
+            );
+
+            container.append(wrapper);
+        }
+        return wrapper;
+    };
+
+    return {
+        display: function () {
+            var container = $(this.container_selector()),
+                wrapper   = template.call(this, container),
+                isEmpty   = this.is_empty();
+
+            container.toggleClass("empty", isEmpty)
+
+            wrapper.height(this.height);
+
+            if (isEmpty) {
```

Comment:
> and have display take an explicit parameter instead?

I agree 👍 looks better.

> This can be simplified as wrapper.toggle(isEmpty).

Yes. 😅 didn't really spot it because the first code was a little bit more complicated than what it currently is.